### ISSUE TITLE
Can you offer some example using maven-atg-assembler-plugin

### ISSUE THREAD

**Issue by hanrw** [tacitknowledge/maven-atg-assembler-plugin#2]
Hi, i'm trying to using maven-atg-assembler-plugin to bulid atg project. but there is no more details about project hierarchy and how to config maven-atg-assembler-plugin, can you public a demo for us?

thank you.

**Comment by scottfromsf** [comment]
@witherspore, do you happen to have an example of this plugin in use anywhere?

**Comment by witherspore** [comment]
we used it last in the sephora code.  Don't have that anymore

**Comment by hanrw** [comment]
thank you for response, this is my pom file
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="
http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

```
<groupId>atg-mvn-quickstart</groupId>
<artifactId>madeinchina</artifactId>
<version>0.0.1-SNAPSHOT</version>
<packaging>war</packaging>

<name>atg maven quickstart</name>
<url>http://maven.apache.org</url>

<properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
</properties>

<dependencies>
    <dependency>
        <groupId>junit</groupId>
        <artifactId>junit</artifactId>
        <version>3.8.1</version>
        <scope>test</scope>
    </dependency>
</dependencies>
<build>
    <plugins>
        <plugin>
            <groupId>com.tacitknowledge.maven.plugins</groupId>
            <artifactId>tk-maven-atg-assembler-plugin</artifactId>
            <version>10.2</version>
            <executions>
                <execution>
                    <configuration>
                        <dynamoRoot>D:/ATG</dynamoRoot>
                        <requiredModules>
                            DSS,DAS
                        </requiredModules>
                    </configuration>
                    <goals>
                        <goal>
                      assemble-unpacked-ear
                    </goal>
                    </goals>
                </execution>
            </executions>

        </plugin>
        <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-compiler-plugin</artifactId>
            <version>2.0</version>
            <configuration>
                <source>1.4</source>
                <target>1.4</target>
            </configuration>
        </plugin>
    </plugins>
</build>
```

</project>

but it not work.

**Comment by hanrw** [comment]
I modifid the source code
add phase  and requiresDependencyResolution  annotation
- @goal ear-manifest
- @phase generate-sources
- @requiresDependencyResolution runtime
  */
  public class EarManifestMojo extends AbstractManifestAssembler
  {

then modify pom like  this
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="
http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

```
<groupId>atg-mvn-quickstart</groupId>
<artifactId>madeinchina</artifactId>
<version>0.0.1-SNAPSHOT</version>
<packaging>pom</packaging>

<name>atg maven quickstart</name>
<url>http://maven.apache.org</url>

<properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
</properties>

<dependencies>
    <dependency>
        <groupId>junit</groupId>
        <artifactId>junit</artifactId>
        <version>3.8.1</version>
        <scope>test</scope>
    </dependency>
</dependencies>
<build>
    <plugins>
        <plugin>
            <groupId>com.tacitknowledge.maven.plugins</groupId>
            <artifactId>tk-maven-atg-assembler-plugin</artifactId>
            <version>10.2</version>
            <executions>
                <execution>
                    <id>generate-sources</id>
                    <phase>generate-sources</phase>
                    <configuration>
```

<dynamoModules>atg-maven-quickstart</dynamoModules>
                            <destinationFile>D:/ATG</destinationFile>
                            <dynamoRoot>D:/ATG</dynamoRoot>
                            <requiredModules>
                                DSS,DAS
                            </requiredModules>
                        </configuration>
                        <goals>
                            <goal>
                          assemble-unpacked-ear
                        </goal>
                        </goals>
                    </execution>
                </executions>

```
        </plugin>
        <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-compiler-plugin</artifactId>
            <version>2.0</version>
            <configuration>
                <source>1.4</source>
                <target>1.4</target>
            </configuration>
        </plugin>
    </plugins>
</build>
```

</project>

it can works, but failed
 (generate-sources) on project madeinchina: can't resolve method:
atg.appassembly.ant.CreateUnpackedEarTask.setOmitLicenses(boolean) -> [Help
1]

**Comment by hanrw** [comment]
remove this configure(eartask, "omitLicenses"); line
in CreateUnpackedEarMojo it works for atg10.2.it just can bulid. if you
give a sample is much better.
thank you.

**Comment by scottfromsf** [comment]
Hi @MadeInChina, are ou having specific issues with the plugin or are you looking where to start with it? We've not used this in a number of years but are looking for some code in our archives that uses it.

**Comment by hanrw** [comment]
yes i'm trying to using this plugin to bulid atg10.2. maybe can create a
maven archetype.

what i am doing is that  the project has three module

atg-maven-archetype
  |------atg-maven-core (the core java code)

  |------atg-maven-web(the web application)

  |------atg-maven-packing(this module is using your plugin to build
atg-maven-core and atg-maven-web to ear )

 the ear structure should like this

 atg-maven-archetype.ear
   |------atg_bootstrap.war
   |------atg_bootstrap_ejb.jar
   |------atglib
   |------atg-maven-web.war
   |------META-INF
 this is what i wanna to do.but actually i can't packing atg-maven-web into
atg-maven-archetype.ear

**Comment by witherspore** [comment]
@MadeInChina

If its okay, would you send over your pom.xml files for archetype, core,
web, and packing?  I suspect there's a missing configuration setting that
would link the web module into the rest.

Does the unpacked ear have everything but the web module?

Most of the mojos in this plugin focus on generating the manifest files.
 It just leverages the atg assembler ant task for creating the ear.

Matthew

**Comment by hanrw** [comment]
okay,i send the project to you.

**Comment by hanrw** [comment]
Hi @witherspore,did you found any configuration to link web module?

**Comment by witherspore** [comment]
@MadeInChina

I didn't see any project (pom) files attached.  Directory structure would
be helpful too.  Would you send them directly to my email address at
witherspore@gmail.com?

Github may have filtered those out.

Thanks

**Comment by hanrw** [comment]
@witherspore,just send to your email. you can check it.

Thx.
